compiler/rustc_codegen_ssa/src/lib.rs

```diff
@@ -146,8 +145,6 @@ pub struct CrateInfo {
     pub used_crate_source: FxHashMap<CrateNum, Lrc<CrateSource>>,
     pub used_crates_static: Vec<(CrateNum, LibSource)>,
     pub used_crates_dynamic: Vec<(CrateNum, LibSource)>,
-    pub lang_item_to_crate: FxHashMap<LangItem, CrateNum>,
-    pub missing_lang_items: FxHashMap<CrateNum, Vec<LangItem>>,
```

Comment:
:sparkling_heart:

Reply:
I'd be interested in getting rid of this without wrapping all of Rust crates into a group.

Reply:
I think the only way to do that would be to remove the concept of weak lang items. (lang items that don't have to be defined yet when using them) We can't do this however as `#[panic_handler]` is a weak lang item.

Reply:
Ah, I see. I didn't know that was possible. Seems that ld has `--require-defined=` which is like `-u` except that the symbol has to be defined in one of the input objects and can't be left undefined in the output object.

Reply:
Only ld.bfd has `--require-defined`. ld.gold only has `-u` just like macOS's ld and other Unix linkers it seems. link.exe and lld don't need linker groups.

Reply:
Now that we have `symbols.o` (https://github.com/rust-lang/rust/pull/95604) it may be an even better way to do stuff like this.

Reply:
This suggestion is now implemented in https://github.com/rust-lang/rust/pull/100404.